monitoring/alerts/spark-rules.yml

```diff
@@ -0,0 +1,98 @@
+groups:
+- name: Spark_Alerts
+  rules:
+    - alert: Spark_Dispatcher_Count
+      expr: count(spark_mesos_cluster_driver_launched) >= 50
+      for: 1m
+      labels:
+        severity: informational
+      annotations:
+        description: 'Spark has reached desired number of dispatchers'
+
+    - alert: Spark_Dispatcher_Count_Decreasing
+      expr: count(spark_mesos_cluster_driver_launched offset 5m) - count(spark_mesos_cluster_driver_launched) > 0
+      for: 1m
+      labels:
+        severity: major
+      annotations:
+        description: 'Spark dispatcher count unexpectedly decreasing'
+
+    - alert: Spark_Job_Count
+      expr: count(spark_mesos_cluster_executor_count) >= 1000
+      for: 1m
+      labels:
+        severity: informational
+      annotations:
+        description: 'Spark has reached desired number of jobs'
+
+    - alert: Spark_Job_Count_Decreasing
+      expr: count(spark_mesos_cluster_executor_count offset 5m) - count(spark_mesos_cluster_executor_count) > 0
+      for: 1m
+      labels:
+        severity: major
+      annotations:
+        description: 'Spark job count unexpectedly decreasing'
+
+    - alert: Spark_CPU_Job
+      expr: count(spark_mesos_cluster_executor_count{task_name=~".*monte.*"}) >= 750
+      for: 1m
+      labels:
+        severity: informational
+      annotations:
+        description: 'Spark has reached desired number of CPU jobs'
+
+    - alert: Spark_GPU_Jobs
+      expr: count(spark_mesos_cluster_executor_count{task_name=~".*image.*"}) >= 10
+      for: 1m
+      labels:
+        severity: informational
+      annotations:
+        description: 'Spark has reached desired number of GPU jobs'
+
+    - alert: Spark_Streaming_Jobs
+      expr: count(spark_mesos_cluster_executor_count{task_name=~".*Kafka.*"}) >= 250
+      for: 1m
+      labels:
+        severity: informational
+      annotations:
+        description: 'Spark has reached desired number of Streaming jobs'
+    
+    - alert: Spark_Average_Queue_Depth
+      expr: avg(spark_mesos_cluster_driver_waiting) >= 50
+      for: 3m
+      labels:
+        severity: major
+      annotations:
+        description: 'Average number of spark jobs queued for each dispatcher growing rapidly, indicating slower launch rates for new jobs or current jobs taking longer to complete'
```

Comment:
The message suggests that you're measuring the rate of queue length growth, while in fact the expression just looks at queue length in absolute terms. Moreover, the average is not the best indicator. Please change it to `max()` or 99th percentile, and change the message to say something like "[Longest/99th %ile] spark job queue is over X times longer than expected, indicating..."

Reply:
Actually, after re-reviewing the ticket it looks like product wanted:
> MUST have a queue depth of between 50 and 200 drivers for 75% of the overall test duration per Dispatcher

So I'm changing the alert to an individual value over 200 queued jobs to indicate that it is beyond product limits.